src/map/label.cpp

```diff
@@ -145,60 +145,60 @@ void map_labels::set_team(const team* team)
-	// (We do not use get_label_private() here because we might need
-	// the label_map as well as the terrain_label.)
-	team_label_map::iterator current_label_map = labels_.find(team_name);
-	label_map::iterator current_label;
-
-	if(current_label_map != labels_.end() &&
-		(current_label = current_label_map->second.find(loc)) != current_label_map->second.end())
-	{
-		// Found old checking if need to erase it
-		if(text.str().empty()) {
-			// Erase the old label.
-			current_label_map->second.erase(current_label);
-
-			// Restore the global label in the same spot, if any.
-			if(terrain_label* global_label = get_label_private(loc, "")) {
-				global_label->recalculate();
-			}
-		} else {
-			current_label->second.update_info(
-				text, creator, tooltip, team_name, color, visible_in_fog, visible_in_shroud, immutable, category);
-
-			res = &current_label->second;
-		}
-	} else if(!text.str().empty()) {
-		// See if we will be replacing a global label.
-		terrain_label* global_label = get_label_private(loc, "");
-
-		// Add the new label.
-		res = add_label(
-			*this, text, creator, team_name, loc, color, visible_in_fog, visible_in_shroud, immutable, category, tooltip);
-
-		// Hide the old label.
-		if(global_label != nullptr) {
-			global_label->recalculate();
-		}
-	}
-
-	categories_dirty = true;
-	return res;
+    const t_string& text,
+    const int creator,
+    const std::string& team_name,
+    const color_t color,
+    const bool visible_in_fog,
+    const bool visible_in_shroud,
+    const bool immutable,
+    const std::string& category,
+    const t_string& tooltip)
+{
+    terrain_label* res = nullptr;
+
+    // See if there is already a label in this location for this team.
+    team_label_map::iterator current_label_map = labels_.find(team_name);
+    label_map::iterator current_label;
+
+    if (current_label_map != labels_.end() &&
+        (current_label = current_label_map->second.find(loc)) != current_label_map->second.end())
+    {
+        // Found old label, checking if it needs to be erased
+        if (text.str().empty()) {
+            // Erase the old label.
+            current_label_map->second.erase(current_label);
+        }
+        else {
+            current_label->second.update_info(
+                text, creator, tooltip, team_name, color, visible_in_fog, visible_in_shroud, immutable, category);
+            res = &current_label->second;
+        }
+    }
+    else if (!text.str().empty()) {
+        // Add the new label.
+        res = add_label(
+            *this, text, creator, team_name, loc, color, visible_in_fog, visible_in_shroud, immutable, category, tooltip);
+    }
+
+    // Check if it's a global label (team_name is empty) and apply the update to all labels.
+    if (team_name.empty()) {
+        for (auto& team_labels : labels_) {
+            if (team_labels.first != team_name) {
+                terrain_label* global_label = team_labels.second[loc];
+                if (global_label) {
```

Comment:
I'm pretty sure our style guide dictates no space between if/for and the opening parenthesis.

Reply:
I am working on it. Give me some time.